pkg/networkservice/chains/nsmgr/single_test.go

```diff
@@ -119,7 +119,8 @@ func Test_ShouldCorrectlyAddForwardersWithSameNames(t *testing.T) {
 	require.NoError(t, err)
 
 	forwarderReg := &registry.NetworkServiceEndpoint{
-		Name: "forwarder",
+		Name:                "forwarder",
+		NetworkServiceNames: []string{"forwarder"},
```

Comment:
@edwarnicke 
1. Should we add const for forwarder networkservicename into api? 
2. Should we create a registry network service endpoint chain element for registration forwarders?

Reply:
> Should we create a registry network service endpoint client chain element for registration forwarders?

I don't quite follow this, could you say more?

Reply:
Sure, we can add something like:

```go
type setForwarderNetworkServiceNameNSEClient struct{} 

func (n *setNetworkServiceNamesNSEClient) Register(ctx context.Context, in *registry.NetworkServiceEndpoint, opts ...grpc.CallOption) (*registry.NetworkServiceEndpoint, error) {
	in.NetworkServiceNames = []string{"forwarder"}
...
```

This client can be used by forwarders on registration.

Reply:
Ah... that's interesting... but how does it handle the case of that being overriden for some reason by env variables in the forwarder?

Better question... where is the 'forwarder' string encoded in the NSMgr ?

Reply:
> Better question... where is the 'forwarder' string encoded in the NSMgr ?

See at discovercrossnse chain element.

Reply:
@edwarnicke What about the first question?

Reply:
I'd hold off on adding a constant to the API currently.  I'm still curious how:

'forwarder' gets set as the magic string in NSMgr... I'd be more inclined to make it a settable parameter via an option and have it flow through from the top of the cmd-nsmgr ...

Reply:
>'forwarder' gets set as the magic string in NSMgr... I'd be more inclined to make it a settable parameter via an option and have it flow through from the top of the cmd-nsmgr ...

Also all forwarders should know about the constant to be correctly registered.

Reply:
if you are fine with this, then I'll add the option as you suggested.

Reply:
Forwarders should already be able to overload the NS name via ENV variable... do they not?

Reply:
Yes, but all forwarders and nsmgr on the cluster should be synchronized with each other.

Reply:
Mostly, I like your suggestion, but just shared another possible way

Reply:
Fixed in https://github.com/networkservicemesh/sdk/pull/1122/commits/1a0be07e86e600742b488cdc1531b329d8025d00